File: politiks-bff-nest/src/dtos/partido.dto.ts
import { IsNotEmpty, IsString, IsOptional, IsNumber, MinLength, MaxLength } from 'class-validator';
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

/**
 * DTO para transferência de dados de Partido.
 * Segue o padrão DTO (Data Transfer Object) para desacoplar a API da camada de domínio.
 */
export class PartidoDto {
  @ApiPropertyOptional({ description: 'ID do partido', example: 1 })
  @IsOptional()
  @IsNumber()
  id?: number;

  @ApiProperty({ description: 'Nome do partido', example: 'Partido Exemplo', minLength: 2, maxLength: 100 })
  @IsNotEmpty({ message: 'O nome do partido é obrigatório' })
  @IsString({ message: 'O nome deve ser uma string' })
  @MinLength(2, { message: 'O nome do partido deve ter pelo menos 2 caracteres' })
  @MaxLength(100, { message: 'O nome do partido deve ter no máximo 100 caracteres' })
  nome: string;

  @ApiProperty({ description: 'Sigla do partido', example: 'PEX', minLength: 2, maxLength: 20 })
  @IsNotEmpty({ message: 'A sigla do partido é obrigatória' })
  @IsString({ message: 'A sigla deve ser uma string' })
  @MinLength(2, { message: 'A sigla do partido deve ter pelo menos 2 caracteres' })
  @MaxLength(20, { message: 'A sigla do partido deve ter no máximo 20 caracteres' })
  sigla: string;

  @ApiPropertyOptional({ description: 'Descrição do partido', example: 'Partido fundado em...' })
  @IsOptional()
  @IsString({ message: 'A descrição deve ser uma string' })
  descricao?: string;

  @ApiPropertyOptional({ description: 'Data de fundação do partido', example: '1980-01-01' })
  @IsOptional()
  @IsString({ message: 'A data de fundação deve ser uma string' })
  dataFundacao?: string;
}

File: politiks-api/src/main/java/org/politiks/model/dto/PartidoDTO.java
package org.politiks.model.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/**
 * DTO para transferência de dados de Partido entre a API e o cliente.
 * Segue o padrão DTO para desacoplar a API da camada de domínio.
 */
public class PartidoDTO {
    
    private Long id;
    
    @NotBlank(message = "O nome do partido é obrigatório")
    @Size(min = 2, max = 100, message = "O nome do partido deve ter entre 2 e 100 caracteres")
    private String nome;
    
    @NotBlank(message = "A sigla do partido é obrigatória")
    @Size(min = 2, max = 20, message = "A sigla do partido deve ter entre 2 e 20 caracteres")
    private String sigla;
    
    private String descricao;
    
    private String dataFundacao;
    
    // Construtores
    public PartidoDTO() {
    }
    
    public PartidoDTO(Long id, String nome, String sigla, String descricao, String dataFundacao) {
        this.id = id;
        this.nome = nome;
        this.sigla = sigla;
        this.descricao = descricao;
        this.dataFundacao = dataFundacao;
    }
    
    // Getters e Setters
    public Long getId() {
        return id;
    }
    
    public void setId(Long id) {
        this.id = id;
    }
    
    public String getNome() {
        return nome;
    }
    
    public void setNome(String nome) {
        this.nome = nome;
    }
    
    public String getSigla() {
        return sigla;
    }
    
    public void setSigla(String sigla) {
        this.sigla = sigla;
    }
    
    public String getDescricao() {
        return descricao;
    }
    
    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }
    
    public String getDataFundacao() {
        return dataFundacao;
    }
    
    public void setDataFundacao(String dataFundacao) {
        this.dataFundacao = dataFundacao;
    }
}
